### PR TITLE
Initial Product catalogue endpoint (paging + filter by category)

### PR DESCRIPTION
**Pull request by campingtree**
* Added 3 endpoints:
    * to reach all paged products + filter + sorting
    * to get a single product (with additional info)
    * to get reviews by product id
* Installed and configured Automapper
* Installed and configured FluentValidations
* Added initial global error handling middleware
* Moved some properties from `ProductInfo` to `Product` entity.
* Made dbContext transient, because it's not thread safe
* CORS to allow cross-origin requests with * (for develop). Will make it explicit for deployed env.

### CODE REVIEW

File: src/Mantasflowers.Contracts/Product/Response/GetDetailedProductResponse.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+{
+    public class GetDetailedProductResponse
+    {
+        public Guid Id { get; set; }
+
+        public string Name { get; set; }
+
+        public ProductCategory Category { get; set; }
+
+        public string ShortDescription { get; set; }
```
Comment (alwaysintune) [inline src/Mantasflowers.Contracts/Product/Response/GetDetailedProductResponse.cs]: No need for
```cs
public string ShortDescription { get; set; }
```
when there is whole 
```cs
public string Description { get; set; }
```
 in a detailed product response.

File: src/Mantasflowers.Contracts/Review/GetProductReviewsResponse.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+using System;
+
+namespace Mantasflowers.Contracts.Review
+{
+    public class GetProductReviewsResponse
+    {
+        public Guid ProductId { get; set; }
+
+        public Guid UserId { get; set; }
```
Comment (alwaysintune) [inline src/Mantasflowers.Contracts/Review/GetProductReviewsResponse.cs]: There's no need to send user id back. Also, noticed there's a review (submission/creation) date missing.

**Comment by alwaysintune** [comment]
Common package is name as "Commom"

File: src/Mantasflowers.Contracts/Review/GetProductReviewsResponse.cs (hunk 2 above)
Comment (campingtree) [reply src/Mantasflowers.Contracts/Review/GetProductReviewsResponse.cs]: I mapped the date from `UpdatedOn`.

**Review by alwaysintune — APPROVED** [review]

**Pull request merged**